### PR TITLE
feat(gcs): add property skip_credential

### PR DESCRIPTION
**Pull request by sanadhis**
Follow-up on #837 

Turns out specifying no credential **is not equal** with adding `option.WithoutAuthentication()` to gcloud SDK. 

My setup is to have this clickhouse-backup runs as side container next to Clickhouse, configured with k8s service account that has proper workload-identity-federation in GKE, see [here](https://cloud.google.com/kubernetes-engine/docs/concepts/workload-identity#credential-flow).

If we set `option.WithoutAuthentication()`, ADC wouldn't even bother to ask **metadata server**, which makes the auth fails for K8s service account. See [here](https://cloud.google.com/docs/authentication/application-default-credentials#attached-sa)

error observed in k8s:
```
2024/02/21 18:27:12.908155  info SELECT macro, substitution FROM system.macros logger=clickhouse
2024/02/21 18:27:12.994845  warn BackupList bd.Walk return error: googleapi: Error 401: Anonymous caller does not have storage.objects.list access to the Google Cloud Storage bucket. Permission 'storage.objects.list' denied on resource (or it may not exist)., required logger=gcs
```

test from local:
```
2024/02/21 20:30:36.631375  info docker exec clickhouse ls -1 /var/lib/clickhouse/backup/*TestIntegrationGCSWithCustomEndpoint*
2024/02/21 20:30:36.759534  info Drop all databases       
2024/02/21 20:30:36.912380  info docker exec gcs sh -c ls -lh /data/altinity-qa-test/
--- PASS: TestIntegrationGCSWithCustomEndpoint (79.47s)
PASS
ok      command-line-arguments  79.848s
```

### CODE REVIEW

**Comment by Slach** [comment]
to fix tests please run

```
# carefull it delete all containers
for id in $(docker ps -a -q); do docker stop "$id"; docker rm -f "$id"; done;
rm -fv ./test/testflows/clickhouse_backup/tests/snapshots/cli.py.cli.snapshot
cp -fv ./test/testflows/.env.example ./test/testflows/.env
# provide actual credentials
mcedit ./test/testflows/.env 
RUN_TESTS="/clickhouse backup/cli/default config" ./test/testflows/run.sh
git add ./test/testflows/clickhouse_backup/tests/snapshots/cli.py.cli.snapshot
git commit -s -m "update snapshot"
git push
```

**Comment by sanadhis** [comment]
> to fix tests please run
> 
> ```
> # carefull it delete all containers
> for id in $(docker ps -a -q); do docker stop "$id"; docker rm -f "$id"; done;
> rm -fv ./test/testflows/clickhouse_backup/tests/snapshots/cli.py.cli.snapshot
> cp -fv ./test/testflows/.env.example ./test/testflows/.env
> # provide actual credentials
> mcedit ./test/testflows/.env 
> RUN_TESTS="/clickhouse backup/cli/default config" ./test/testflows/run.sh
> git add ./test/testflows/clickhouse_backup/tests/snapshots/cli.py.cli.snapshot
> git commit -s -m "update snapshot"
> git push
> ```

Hey, the test pass suites pass but doesn't re-generate `./test/testflows/clickhouse_backup/tests/snapshots/cli.py.cli.snapshot`

```
Passing

✔ [ OK ] /clickhouse backup/cli (166ms)
✔ [ OK ] /clickhouse backup (50s 758ms)

Coverage

QA-SRS013 ClickHouse Backup Utility
  72 requirements (2 satisfied 2.8%, 70 untested 97.2%)

1 suite (1 ok)
2 scenarios (2 ok)
43 steps (43 ok)

Total time 50s 758ms

Executed on Feb 22,2024 7:50
TestFlows.com Open-Source Software Testing Framework v1.9.230705.1122619
```

I am not sure if it's okay to remove this `./test/testflows/clickhouse_backup/tests/snapshots/cli.py.cli.snapshot`

**Comment by Slach** [comment]
you deleted `cli.py.cli.snapshot` instead of patch it
try  just run `./test/testflows/run.sh`

File: test/testflows/clickhouse_backup/tests/snapshots/cli.py.cli.snapshot [hunk 1]
```diff
@@ -1,6 +0,0 @@
...lete_resumable_after_restart: true\', \'ftp:\', \'    address: ""\', \'    timeout: 2m\', \'    username: ""\', \'    password: ""\', \'    tls: false\', \'    skip_tls_verify: false\', \'    path: ""\', \'    object_disk_path: ""\', \'    compression_format: tar\', \'    compression_level: 1\', \'    debug: false\', \'sftp:\', \'    address: ""\', \'    port: 22\', \'    username: ""\', \'    password: ""\', \'    key: ""\', \'    path: ""\', \'    object_disk_path: ""\', \'    compression_format: tar\', \'    compression_level: 1\', \'    debug: false\', \'azblob:\', \'    endpoint_schema: https\', \'    endpoint_suffix: core.windows.net\', \'    account_name: ""\', \'    account_key: ""\', \'    sas: ""\', \'    use_managed_identity: false\', \'    container: ""\', \'    path: ""\', \'    object_disk_path: ""\', \'    compression_level: 1\', \'    compression_format: tar\', \'    sse_key: ""\', \'    buffer_size: 0\', \'    buffer_count: 3\', \'    timeout: 4h\', \'    debug: false\', \'custom:\', \'    upload_command: ""\', \'    download_command: ""\', \'    list_command: ""\', \'    delete_command: ""\', \'    command_timeout: 4h\', \'    commandtimeoutduration: 4h0m0s\']'"""
```
Comment (Slach) [inline test/testflows/clickhouse_backup/tests/snapshots/cli.py.cli.snapshot:1]: please rollback these changes

**Review by Slach — CHANGES_REQUESTED** [review]

**Comment by Slach** [comment]
ok. let me merge your changes, will restore snapshot from my side

**Comment by sanadhis** [comment]
@Slach thanks a lot! Sorry for the regression. Appreciate your fast response.

**Pull request merged**